src/main/java/appeng/parts/p2p/PartP2PFluids.java

```diff
@@ -183,33 +232,43 @@ public int fill( final EnumFacing from, final FluidStack resource, final boolean
 		return used;
 	}
 
-	private Stack<PartP2PLiquids> getDepth()
+	@Override
+	public FluidStack drain( FluidStack resource, boolean doDrain )
 	{
-		Stack<PartP2PLiquids> s = DEPTH.get();
+		return null;
+	}
+
+	@Override
+	public FluidStack drain( int maxDrain, boolean doDrain )
+	{
+		return null;
+	}
+
+	private Stack<PartP2PFluids> getDepth()
+	{
+		Stack<PartP2PFluids> s = DEPTH.get();
 
 		if( s == null )
 		{
-			DEPTH.set( s = new Stack<PartP2PLiquids>() );
+			DEPTH.set( s = new Stack<PartP2PFluids>() );
 		}
 
 		return s;
 	}
 
-	private List<PartP2PLiquids> getOutputs( final Fluid input )
+	private List<PartP2PFluids> getOutputs( final Fluid input )
 	{
-		final List<PartP2PLiquids> outs = new LinkedList<PartP2PLiquids>();
+		final List<PartP2PFluids> outs = new LinkedList<PartP2PFluids>();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Replace the type specification in this constructor call with the diamond operator ("<>"). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2293)